src/layouts/ChatsLayout/components/TheCardGroups/index.vue

```diff
@@ -254,45 +307,177 @@ export default {
         this.searchForMoreRooms(true);
       }
     },
+
+    openModalQueue() {
+      this.showModalQueue = true;
+      this.getListQueues();
+      this.permissionQueuesOptions = [
+        {
+          value: '',
+          label: this.$t('chats.select_your_queues'),
+        },
+      ];
+      this.permissionQueues = [];
+    },
+
+    closeModalQueue() {
+      this.showModalQueue = false;
+      this.permissionQueues = [];
+      this.permissionQueuesOptions = [
+        {
+          value: '',
+          label: this.$t('chats.select_your_queues'),
+        },
+      ];
+    },
+
+    async getListQueues() {
+      try {
+        let me = this.me.email;
+        const response = await Queues.getListQueues(me);
+
+        response.user_permissions.forEach((permission) => {
+          if (permission.role === 1) {
+            this.permissionQueues.push({
+              value: permission.uuid,
+              label: permission.queue_name,
+              role: permission.role,
+            });
+          }
+
+          this.permissionQueuesOptions.push({
+            value: permission.uuid,
+            label: permission.queue_name,
+            role: permission.role,
+          });
+        });
+      } catch (error) {
+        console.error(error);
+      }
+    },
+
+    async saveListQueues() {
+      try {
+        const options = this.permissionQueuesOptions.map(
+          (queue) => queue.value,
+        );
+        const optionsQueues = this.permissionQueues.map((queue) => queue.value);
+        const filter = options.filter(
+          (queue) => !optionsQueues.includes(queue),
+        );
+
+        const selectedQueues = optionsQueues.map((queueUuid) => ({
+          uuid: queueUuid,
+          role: 1,
+        }));
+
+        const unselectedQueues = filter.map((queueUuid) => ({
+          uuid: queueUuid,
+          role: 2,
+        }));
```

Comment:
I advise turning these static "`role`" values into variables, like `const ROLE_ID_SELECTED = 1` and using them here, to make the code more readable.